File: SrdDatabase/ClientApp/src/models/archdeaconry.ts
import { CongregationResults } from "./congregation";
import { ParishResults } from "./parish";
import { PagedParameters, PagedResults, Population } from "./shared";
import { EventResults } from './event';
import { PaymentResults } from "./payment";
import { QuotaResults } from "./quota";

export interface Archdeaconry {
    id?: number;
    name?: string;
    numberOfChristians?: number;
    quota?: number;
    balance?: number;
}

export interface ArchdeaconryDetails {
    archdeaconry: Archdeaconry;
    population: Population;
    parishResults: ParishResults;
    congregationResults: CongregationResults;
    eventResults: EventResults;
    paymentResults: PaymentResults;
    quotaResults: QuotaResults;
}

export interface ArchdeaconryParameters extends PagedParameters {
    name?: string;
}

export interface ArchdeaconryResults extends PagedResults {
    archdeaconries: Archdeaconry[];
}

File: SrdDatabase/Models/Archdeaconries/Archdeaconry.cs
namespace SrdDatabase.Models.Archdeaconries
{
    public class Archdeaconry
    {
        public int Id { get; }

        public string Name { get;  }

        public int? NumberOfChristians { get; }

        public long Quota { get; }

        public long Balance { get; }
        
        public Archdeaconry(
            int id,
            string name,
            int? numberOfChristians,
            int quota,
            int balance)
        {
            Id = id;
            Name = name;
            NumberOfChristians = numberOfChristians;
            Quota = quota;
            Balance = balance;
        }

        // for Dapper
        public Archdeaconry()
        {
        }
    }
}
